self-hosting/README.md

```diff
@@ -0,0 +1,304 @@
+
+```yaml
+version: '3.7'
+
+services:
+  medic-os:
+    container_name: medic-os
+    image: medicmobile/medic-os:cht-3.9.0-rc.2
+    volumes:
+      - medic-data:/srv
+    ports:
+     - 80:80
+     - 443:443
+    working_dir: /srv
+    depends_on:
+      - haproxy
+    networks:
+      - medic-net
+    environment:
+      - DOCKER_NETWORK_NAME=haproxy
+      - DOCKER_COUCHDB_ADMIN_PASSWORD=$DOCKER_COUCHDB_ADMIN_PASSWORD
+
+  haproxy:
+    container_name: haproxy
+    image: medicmobile/haproxy:rc-1.17
+    volumes:
+      - medic-data:/srv
+    environment:
+      - COUCHDB_HOST=medic-os
+      - HA_PASSWORD=$DOCKER_COUCHDB_ADMIN_PASSWORD
+    networks:
+      - medic-net
+
+volumes:
+  medic-data:
+    name: medic-data
+
+networks:
+  medic-net:
+    name: medic-net
+
+```
+
+The install requires an admin password that it will configure in the database. You need to provide this externally as an environment variable. before you run the compose file, you need to export this variable as shown below.
+
+`export DOCKER_COUCHDB_ADMIN_PASSWORD=myAwesomeCHTAdminPassword`
+
+You can then run docker-compose in the folder where you put your compose configuration yaml file as shown below.
+
+```bash
+## run compose inside the coconfigulation folder as root 
+
+$ sudo docker-compose up 
+
+## In detached mode 
+
+$ sudo docker-compose up  -d 
+
+##  Outside the config folder 
+
+$ sudo docker-compose -f /path/to/docker-compose.yml up -d
+
+```
+
+Note In certain shells, docker-compose may not interpolate the admin password that was exported above. In that case, your admin user had a password automatically generated. Note the New CouchDB Administrative User and New CouchDB Administrative Password in the output terminal. You can retrieve these via running `docker logs medic-os` and searching the terminal.
+
+Once containers are setup, please run the following command from your host terminal:
+
+```bash
+
+$ docker exec -it medic-os /bin/bash -c "sed -i 's/--install=3.9.0/--complete-install/g' /srv/scripts/horticulturalist/postrun/horticulturalist"
+$ docker exec -it medic-os /bin/bash -c "/boot/svc-disable medic-core openssh && /boot/svc-disable medic-rdbms && /boot/svc-disable medic-couch2pg"
+
+```
+
+The first command fixes a postrun script for horticulturalist to prevent unique scenarios of re-install. The second command removes extra services that you will not need.
+
+### Visit your project
+
+Open a browser to: https://localhost
+
+You will have to click to through the SSL Security warning. Click Advanced -> Continue to site.
+
+
+### Clean up an re-install
+
+You could have missed some of the instructions above and end with a bricked setup.  You can use the commands below to clean up and start afresh.
+
+Stop containers:
+
+- `docker-compose down or docker stop medic-os && docker stop haproxy`
```

Comment:
Is the user supposed to run the command: `docker-compose down or docker stop medic-os`? If so, cool! 

If not, and this is a choice between `docker-compose down` or `docker stop`, let's just choose one (there's 100 ways to skin a cat, as we say in the US).  Maybe go with `docker-compose` and then update `docker stop haproxy` to match? This assumes that the haproxy container has a compose.yml - I don't really know since I don't know docker ;)

Reply:
The choice is to use `docker-compose down` from the same directory your `yml` file lives or run both of these commands from with the host system `docker stop medic-os && docker stop haproxy`.    I believe you could also specify the `-f` arg with compose. Comes back to the many ways to skin a cat and how much we want to be involved in training on how docker works. 

The containers we start up as part of our `docker-compose.yml` are medic-os, and haproxy. There isn't a separate `docker-compose.yml` file for haproxy.